jinahub/indexers/searcher/HnswlibSearcher/hnswlib_searcher.py

```diff
@@ -1,128 +1,271 @@
 __copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from typing import Dict, List, Optional
+import json
+from typing import Dict, Iterable, Optional
 
 import hnswlib
-import numpy as np
+from bidict import bidict
 from jina import Document, DocumentArray, Executor, requests
-from jina_commons import get_logger
-from jina_commons.indexers.dump import import_vectors
+from jina.logging.logger import JinaLogger
 
 
 class HnswlibSearcher(Executor):
-    """Hnswlib powered vector indexer
+    """Hnswlib powered vector indexer.
 
-    For more information about the Hnswlib supported parameters, please consult:
-        - https://github.com/nmslib/hnswlib
-
-    .. note::
-        Hnswlib package dependency is only required at the query time.
+    This indexer uses the HNSW algorithm to index and search for vectors. It does not
+    require training, and can be built up incrementally.
     """
 
     def __init__(
         self,
-        default_top_k: int = 10,
+        top_k: int = 10,
         metric: str = 'cosine',
-        dump_path: Optional[str] = None,
-        default_traversal_paths: Optional[List[str]] = None,
-        is_distance: bool = False,
+        dim: int = 0,
+        max_elements: int = 1_000_000,
         ef_construction: int = 400,
         ef_query: int = 50,
         max_connection: int = 64,
+        dump_path: Optional[str] = None,
+        traversal_paths: Iterable[str] = ('r',),
         *args,
         **kwargs,
     ):
         """
-        Initialize an HnswlibSearcher
-
-        :param default_top_k: get tok k vectors
-        :param distance: distance can be 'l2', 'ip', or 'cosine'
-        :param dump_path: the path to load ids and vecs
-        :param traverse_path: traverse path on docs, e.g. ['r'], ['c']
-        :param reverse_score: True if add reversed distance as the `similarity` for match score, else return `distance` as score for match score.
-        :param ef_construction: defines a construction time/accuracy trade-off
-        :param ef_query:  sets the query time accuracy/speed trade-off
-        :param max_connection: defines tha maximum number of outgoing connections in the graph
+        :param top_k: Number of results to get for each query document in search
+        :param distance: Distance type, can be 'l2', 'ip', or 'cosine'
```

Comment:
metric or distance

Reply:
Actually, distance is always returned, see Actually, distance is always returned, see https://github.com/nmslib/hnswlib#supported-distances

Reply:
No, the name is different from the one in the constructor

Reply:
Ah ok, got it

Reply:
Renamed all to `distance`